src/NuGet.Core/NuGet.Packaging/Rules/IconAndIconUrlUsageWarning.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Rules
+{
+    public class IconAndIconUrlUsageWarning : IPackageRule
+    {
+        public string MessageFormat { get; }
+
+        public IconAndIconUrlUsageWarning(string messageFormat)
+        {
+            MessageFormat = messageFormat ?? throw new ArgumentNullException(nameof(messageFormat));
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var nuspecReader = builder?.NuspecReader;
```

Comment:
When can this be null?

Reply:
When you call this Validation Rule outside of the NuGet world, i.e., in an external library.

Reply:
What's the pointing of a null reader? It's an invalid call from an external. 

We get to set the rules when writing APIs, and the rule is having a valid reader.

None of our other rules do that, https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Rules/InvalidFrameworkFolderRule.cs

Reply:
Ok. Will change that. Should I do the same for the NuspecReader properties below that line?

Reply:
yup.

Reply:
Will change `IconUrlDeprecationWarning`. This class is removed in the next PR: https://github.com/NuGet/NuGet.Client/pull/3006